src/inmanta/data/__init__.py

```diff
@@ -735,26 +741,38 @@ def __init__(
         self,
         field_type: Type[T],
         required: bool = False,
-        unique: bool = False,
-        reference: bool = False,
```

Comment:
I'm not familiar with these parameters so I can't really review this part.

Reply:
This was a leftover from mongodb, it was not used anywhere anymore. So it can go.

- unique: a constraint that is now enforced by indexes in the db which mongo did not have
- reference: this meant that we needed to store the id instead of just the entire dict